packages/core/src/render3/instructions.ts

```diff
@@ -139,13 +131,13 @@ export function enterView(newViewState: ViewState, host: LElement | LView | null
 
 export const leaveView: (newViewState: ViewState) => void = enterView as any;
 
-export function createViewState(viewId: number, renderer: Renderer3): ViewState {
+export function createViewState(viewId: number, renderer: Renderer3, ngStaticData: NgStaticData): ViewState {
   const newView = {
     parent: currentView,
     id: viewId,    // -1 for component views
     node: null !,  // until we initialize it in createNode.
     data: null !,  // Hack use as a marker for creationMode
-    directives: [],
+    ngStaticData,
```

Comment:
Not a fan of `ngStaticData` since renaming the function parameter will break the shape of the object. Could we write `ngStaticData: ngStaticData`.